syft/frameworks/torch/tensors/interpreters/promise.py

```diff
@@ -0,0 +1,185 @@
+        #     self._grad = PromiseTensor(shape=self._shape, tensor_type=self.torch_type()).wrap()
+        #
+        # return self._grad
+
+    def on(self, tensor: "AbstractTensor", wrap: bool = True) -> "AbstractTensor":
+        """
+        Add a syft(log) tensor on top of the tensor.
+
+        Args:
+            tensor: the tensor to extend
+            wrap: if true, add the syft tensor between the wrapper
+            and the rest of the chain. If false, just add it at the top
+
+        Returns:
+            a syft/torch tensor
+        """
+
+        # This is the only difference from AbstractTensor.on()
+        self.obj_type = tensor.type()
+
+        if not wrap:
+
+            self.child = tensor
+
+            return self
+
+        else:
+
+            # if tensor is a wrapper
+            if not hasattr(tensor, "child"):
+                tensor = tensor.wrap()
+
+            self.child = tensor.child
+            tensor.child = self
+
+            tensor.child.parent = weakref.ref(tensor)
+            return tensor
+
+    def __str__(self):
+        return f"[PromiseTensor({self.owner.id}:{self.id}) -future-> {self.obj_type.split('.')[-1]} -blocking-> {len(self.plans)} plans]"
+
+    def __repr__(self):
+        return self.__str__()
+
+    @staticmethod
+    def simplify(self: "PromiseTensor") -> tuple:
+        """Takes the attributes of a FixedPrecisionTensor and saves them in a tuple.
+
+        Args:
+            tensor: a FixedPrecisionTensor.
+
+        Returns:
+            tuple: a tuple holding the unique attributes of the fixed precision tensor.
+        """
+
+        return (
+            sy.serde._simplify(self.id),
+            sy.serde._simplify(self.shape),
+            sy.serde._simplify(self.obj_type),
+            sy.serde._simplify(self.plans),
+        )
+
+    @staticmethod
+    def detail(worker: AbstractWorker, tensor_tuple: tuple) -> "PromiseTensor":
+        """
+            This function reconstructs a FixedPrecisionTensor given it's attributes in form of a tuple.
+            Args:
+                worker: the worker doing the deserialization
+                tensor_tuple: a tuple holding the attributes of the FixedPrecisionTensor
+            Returns:
+                FixedPrecisionTensor: a FixedPrecisionTensor
+            Examples:
+                shared_tensor = detail(data)
+            """
+
+        id, shape, tensor_type, plans = tensor_tuple
+
+        id = sy.serde._detail(worker, id)
+        shape = sy.serde._detail(worker, shape)
+        tensor_type = sy.serde._detail(worker, tensor_type)
+        plans = sy.serde._detail(worker, plans)
+
+        tensor = PromiseTensor(
+            owner=worker, id=id, shape=shape, tensor_type=tensor_type, plans=plans
+        )
+
+        return tensor
+
+
+def CreatePromiseTensor(shape, tensor_type: str, *args, **kwargs):
+    return PromiseTensor(shape, *args, tensor_type=tensor_type, **kwargs).wrap()
+
+
+class Promises:
```

Comment:
Why are these here and should they live in their own file?

Reply:
They are the interface to create promised tensors, the user can do something like `a = sy.Promises.FloatTensor(shape=torch.Size((3, 3)))`.
But I agree, maybe they should be in another file, and we may want to think again about the naming (almost the same as the class Promise). Do you propose something?
@iamtrask